leader-election-impl/src/test/java/com/palantir/leader/LeadershipElectionCheckTest.java

```diff
@@ -71,57 +71,69 @@ public void setup() {
 
-        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
     }
 
     @Test
     public void shouldBeUnhealthyOverallEvenIfIndividualClientsAreHealthy() {
-        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 2,  Duration.ofSeconds(10));
-        markLeaderElectionsAtSpecifiedInterval(CLIENT_2, 3,  Duration.ofSeconds(10));
+        markLeaderElectionsAtSpecifiedInterval(CLIENT_1, 2,  Duration.ofSeconds(10), true);
+        markLeaderElectionsAtSpecifiedInterval(CLIENT_2, 3,  Duration.ofSeconds(10), false);
 
-        assertThat(leaderElectionHealthCheckForOnlyClient1.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheckForOnlyClient1.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
 
-        assertThat(leaderElectionHealthCheckForOnlyClient2.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.HEALTHY);
+        assertThat(leaderElectionHealthCheckForOnlyClient2.leaderElectionRateHealthReport().leaderElectionRate())
+                .isLessThanOrEqualTo(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
 
-        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthStatus())
-                .isEqualTo(LeaderElectionHealthStatus.UNHEALTHY);
+        assertThat(leaderElectionHealthCheck.leaderElectionRateHealthReport().leaderElectionRate())
+                .isGreaterThan(LeaderElectionHealthCheck.MAX_ALLOWED_LAST_5_MINUTE_RATE);
     }
 
     private void markLeaderElectionsAtSpecifiedInterval(Client client,
             int leaderElectionCount,
-            Duration timeIntervalInSeconds) {
+            Duration timeIntervalInSeconds,
+            boolean afterDeactivationPeriod) {
         // The rate is initialized after first tick (5 second interval) of meter with number of marks / interval.
         // Marking before the first interval has passed sets the rate very high, which should not happen in practice.
-        fakeTimeClock.advance(6, TimeUnit.SECONDS);
-        LeaderElectionServiceMetrics metrics = clientLeaderElectionServiceMetricsMap.get(client);
+        if (afterDeactivationPeriod) {
+            fakeTimeClock.advance(14, TimeUnit.MINUTES);
+        }
 
+        LeaderElectionServiceMetrics metrics = clientLeaderElectionServiceMetricsMap.get(client);
         IntStream.range(0, leaderElectionCount).forEach(idx -> {
             metrics.proposedLeadership().mark();
             fakeTimeClock.advance(timeIntervalInSeconds.getSeconds(), TimeUnit.SECONDS);
```

Comment:
Do you want a test for the new behaviour (the clock starts only after first client is registered)?